src/main/java/com/scottlogic/sorts/AuthorPostSorter.java

```diff
@@ -4,36 +4,42 @@
 import com.scottlogic.UserPost;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
-import static java.util.Comparator.*;
-
 public class AuthorPostSorter implements PostSorter {
 
 
     @Override
     public List<UserPost> sort(List<UserPost> inputList, SortOrder orderIn) {
 
         List<UserPost> listToSort = new ArrayList<>();
-        List<UserPost> listSorted = new ArrayList<>();
+        List<UserPost> listSorted;
 
         if (inputList == null) {
             return listToSort;
         }
         listToSort = NullPostChecker.nullPostCheck(inputList);
 
+        listSorted = (orderIn.compareTo(SortOrder.ASC) == 0) ?
+                listToSort.stream().sorted(AuthorName).collect(Collectors.toList()) :
 
-        switch (orderIn) {
-
-            case ASC -> listSorted = listToSort.stream().sorted(Comparator.nullsLast(Comparator.comparing(UserPost::getAuthor, nullsLast(naturalOrder())))).collect(Collectors.toList());
-
+                listToSort.stream().sorted(AuthorName.reversed()).collect(Collectors.toList());
 
-            case DESC -> listSorted = listToSort.stream().sorted(Collections.reverseOrder(comparing(UserPost::getAuthor, nullsFirst(naturalOrder())))).collect(Collectors.toList());
-        }
 
         return listSorted;
     }
+
+    public Comparator<UserPost> AuthorName = (o1, o2) -> {
```

Comment:
You could refactor this to be a private method (returning an int), then construct the Comparator(s) using the method.